### PR TITLE
Added DrawCircle and DrawLine

### PR DESCRIPTION
**Pull request by srandoux**
I did the changes you mentioned.

Regards,

-Sébastien

### CODE REVIEW

**Comment by hzeller** [comment]
Thanks! Much better. Only a few comments above and we are ready to merge.

File: lib/graphics.cc [hunk 1]
```diff
@@ -27,4 +28,70 @@ int DrawText(Canvas *c, const Font &font,
   }
   return x - start_x;
 }
+    
+    
+void DrawCircle(Canvas *c, int32_t x0, int32_t y0, int32_t radius, uint8_t red, uint8_t green, uint8_t blue){
```
Comment (hzeller) [inline lib/graphics.cc]: At the end of this method header, add a space before the curly brace
  `...blue) {`

File: lib/graphics.cc [hunk 2]
```diff
@@ -27,4 +28,70 @@ int DrawText(Canvas *c, const Font &font,
...
+int32_t x = radius, y = 0;
```
Comment (hzeller) [inline lib/graphics.cc]: This indentation looks off in the github diff viewer; they should be indented.

**Comment by hzeller** [comment]
The indentation looks like it is 4 spaces, but this code usually uses 2 spaces. Can you look at that as well ?

**Comment by srandoux** [comment]
I did the updates

**Comment by hzeller** [comment]
Only two little remarks. Will merge this evening (California time).

**Comment by srandoux** [comment]
ok, I have some code for an analog clock and a plasma. I did not add it to your main demo code because it is a big file and i prefered to split it. I committed in on a repo on my github, you can have a look if you want. https://github.com/srandoux/ledmatrix

**Comment by hzeller** [comment]
So the clock and plasma example sound interesting (I didn't compile them, but had a quick look). If you want, you could integrate these as another demo in the demo-main.cc.

This file is getting a bit big, and maybe I am separating these at some point. But having everything for demos in one file is probably as well useful for people that want to get started with copy/paste programming.

**Comment by srandoux** [comment]
Hi, ok I'll do that. maybe not this week but soon. I need to work on antialiased version of the line though.
(the algorithm I know uses alpha channels...). By sorry for the dx==dy==0 case in the line, I forgot It and did not see it happening while testing.

**Pull request merged**